lib/ansible/modules/monitoring/zabbix/zabbix_host.py

```diff
@@ -816,6 +918,14 @@ def main():
             if interface['type'] == 1:
                 ip = interface['ip']
 
+    if macros:
+        # convert macros to zabbix native format - {$MACRO}
+        for macro in macros:
+            if not macro['macro'].startswith('{$'):
+                macro['macro'] = '{$' + macro['macro']
+            if not macro['macro'].endswith('}'):
+                macro['macro'] = macro['macro'] + '}'
+
```

Comment:
Macroname have to be uppercase. Frontend is doing this on-the-flight.

```python
macro['macro'] = macro['macro'].upper()
```

Reply:
Thanks for correcting me. This somehow slipped by me when looking into user macro logic implemented in zabbix_hostmacro module. Should be fixed now.